pkg/pubsub/adapter/converters/scheduler_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/pubsub"
+	cloudevents "github.com/cloudevents/sdk-go"
+	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
+	pubsubcontext "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub/context"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
+)
+
+func TestConvertScheduler(t *testing.T) {
+
+	tests := []struct {
+		name        string
+		message     *cepubsub.Message
+		source      string
+		sendMode    ModeType
+		wantEventFn func() *cloudevents.Event
+		wantErr     string
+	}{{
+		name: "valid attributes",
+		message: &cepubsub.Message{
+			Data: []byte("test data"),
+			Attributes: map[string]string{
+				"jobName":    "projects/knative-gcp-test/locations/us-east4/jobs/cre-scheduler-test",
+				"attribute1": "value1",
+				"attribute2": "value2",
+			},
+		},
+		sendMode: Binary,
```

Comment:
Seems not necessary as every test case uses Binary

Reply:
This is required when we call `Convert(ctx context.Context, msg *cepubsub.Message, sendMode ModeType, converterType string)`